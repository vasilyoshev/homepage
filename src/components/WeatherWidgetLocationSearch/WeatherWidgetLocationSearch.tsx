import React, { useState, ChangeEvent } from 'react';
import { WeatherWidgetLocationSearchProps } from 'interfaces';

export const WeatherWidgetLocationSearch: React.FC<WeatherWidgetLocationSearchProps> = ({ onSearch }) => {
  const [locationSearch, setLocationSearch] = useState('');
  const disableSearch = locationSearch.trim() === '';

  const addLocation = () => {
    onSearch(locationSearch);
    setLocationSearch('');
  };

  const handleInput = (e: ChangeEvent<HTMLInputElement>) => setLocationSearch(e.target.value);

  return (
    <section>
      <label>
        Add Location
        <input type="text" value={locationSearch} onChange={handleInput}/>
      </label>
      <button onClick={addLocation} disabled={disableSearch}>Search</button>
    </section>
  );
};
