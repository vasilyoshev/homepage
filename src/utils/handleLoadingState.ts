import axios from 'axios';
import { decrementLoadingCount, incrementLoadingCount } from 'slices';
import store from 'store';

export const handleLoadingState = (): void => {
  axios.interceptors.response.use(
    (response) => {
      setTimeout(() => store.dispatch(decrementLoadingCount()), 200);

      return response;
    },
    () => {
      store.dispatch(decrementLoadingCount());
    },
  );

  axios.interceptors.request.use((request) => {
    store.dispatch(incrementLoadingCount());

    return request;
  });
};
