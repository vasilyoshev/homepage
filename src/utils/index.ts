export * from './handleLoadingState';
export * from './testUtils';
