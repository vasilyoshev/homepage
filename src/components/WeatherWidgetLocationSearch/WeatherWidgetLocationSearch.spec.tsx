import { render } from 'utils';
import { WeatherWidgetLocationSearch } from './WeatherWidgetLocationSearch';

describe('WeatherWidgetLocationSearch', () => {
  it('renders successfully', () => {
    const { asFragment } = render(<WeatherWidgetLocationSearch onSearch={jest.fn()} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
