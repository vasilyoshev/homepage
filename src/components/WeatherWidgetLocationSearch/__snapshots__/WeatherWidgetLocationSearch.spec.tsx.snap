// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherWidgetLocationSearch renders successfully 1`] = `
<DocumentFragment>
  <section>
    <label>
      Add Location
      <input
        type="text"
        value=""
      />
    </label>
    <button
      disabled=""
    >
      Search
    </button>
  </section>
</DocumentFragment>
`;
